### PR TITLE
feature/registries table

### LINKED ISSUE #30: Add registry table

### LINKED ISSUE #32: Add verification dialog component

### LINKED ISSUE #34: fix code factor found bugs

### PR DESCRIPTION
**Pull request by JustinCast**
# Add Registries table and verification dialog component
- This PR includes new RegistriesTable visualization feature along with verification dialog component.
## How should this be tested
- Clone this branch
- Type `yarn install` command in the terminal
- Type `yarn start` command in the terminal
- Go to http://localhost:3000/registries

### Linked issues
- closes #30
- closes #32
- closes #34

### CODE REVIEW

**Comment by xavier506** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/eoscostarica/notarioeoscr) found multiple issues last seen at 8ab31210f120514d784ed2e5a4383b3a66167ffd:

#### Curly braces are unnecessary here.
[src\components\VerificationDialog\VerificationDialog.js:65
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L65)
#### &#39;handleClose&#39; is missing in props validation
[src\components\VerificationDialog\VerificationDialog.js:35
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L35)
#### &#39;handleCose&#39; PropType is defined but prop is never used
[src\components\VerificationDialog\VerificationDialog.js:171
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L171)
#### Value must be omitted for boolean attributes
[src\components\VerificationDialog\VerificationDialog.js:64
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L64)
#### Missing &quot;key&quot; prop for element in array
[src\routes\Dashboard\DashboardRegistriesTable.js:98
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/routes/Dashboard/DashboardRegistriesTable.js#L98)
#### Identifier &#39;index_position&#39; is not in camel case.
[src\routes\Dashboard\DashboardRegistriesTable.js:34
](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/routes/Dashboard/DashboardRegistriesTable.js#L34)

**Comment by JustinCast** [comment]
> [CodeFactor](https://www.codefactor.io/repository/github/eoscostarica/notarioeoscr) found multiple issues last seen at [8ab3121](https://github.com/eoscostarica/notarioeoscr/commit/8ab31210f120514d784ed2e5a4383b3a66167ffd):
> 
> #### Curly braces are unnecessary here.
> [src\components\VerificationDialog\VerificationDialog.js:65 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L65)
> 
> #### 'handleClose' is missing in props validation
> [src\components\VerificationDialog\VerificationDialog.js:35 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L35)
> 
> #### 'handleCose' PropType is defined but prop is never used
> [src\components\VerificationDialog\VerificationDialog.js:171 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L171)
> 
> #### Value must be omitted for boolean attributes
> [src\components\VerificationDialog\VerificationDialog.js:64 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/components/VerificationDialog/VerificationDialog.js#L64)
> 
> #### Missing "key" prop for element in array
> [src\routes\Dashboard\DashboardRegistriesTable.js:98 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/routes/Dashboard/DashboardRegistriesTable.js#L98)
> 
> #### Identifier 'index_position' is not in camel case.
> [src\routes\Dashboard\DashboardRegistriesTable.js:34 ](https://github.com/eoscostarica/notarioeoscr/blob/8ab31210f120514d784ed2e5a4383b3a66167ffd/src/routes/Dashboard/DashboardRegistriesTable.js#L34)

Fixed!

File: src/components/VerificationDialog/VerificationDialog.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import Slide from '@material-ui/core/Slide'
+import FindInPageIcon from '@material-ui/icons/FindInPage'
+import { rpc } from '../../api/eosjs-api'
+import Typography from '@material-ui/core/Typography'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import styled from '@material-ui/core/styles/styled'
+import Paper from '@material-ui/core/Paper'
+import { useTranslation } from 'react-i18next'
+
+const Transition = React.forwardRef(function Transition(props, ref) {
```
Comment (tetogomez) [inline src/components/VerificationDialog/VerificationDialog.js]: Use `arrow function` and import `forwardRef ` instead to call it directly of `React`

example: 
`const Transition = forwardRef((props, ref) => {...`

File: src/components/VerificationDialog/VerificationDialog.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  borderRadius: '2px',
+  border: '0.5px dashed gray',
+  padding: '3%'
+})
+
+const VerificationDialog = ({ open, handleClose, txData }) => {
+  const [verifying, setVerifying] = useState(false)
+  const [verificationDetails, setVerificationDetails] = useState()
+  const { t } = useTranslation('translations')
+  const verifyTransaction = (tx) => {
```
Comment (tetogomez) [inline src/components/VerificationDialog/VerificationDialog.js:40]: Add a blank space after to variable declaration and before a function

File: src/components/VerificationDialog/VerificationDialog.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      <StyledDialogContent>
+        <Box display="flex" flexDirection="column" justifyContent="start">
+          <Typography>
+            <strong>{t('verificationDialog.title')}</strong> {txData.file}
+          </Typography>
+          <Typography>
+            <strong>{t('verificationDialog.user')}</strong> {txData.account}
+          </Typography>
+          <Typography>
+            <strong>{t('verificationDialog.lastModification')}</strong>{' '}
```
Comment (tetogomez) [inline src/components/VerificationDialog/VerificationDialog.js]: `</strong>{' '}` that space is necessary ?

File: src/components/VerificationDialog/VerificationDialog.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            severity="success"
+          >
+            {verificationDetails.content}
+          </Alert>
+        )}
+        <br />
+        {verificationDetails && verificationDetails.severity === 'success' && (
+          <MessagesBox elevation={3}>
+            <Typography>
+              <strong>{t('registriesTable.blockNum')}:</strong>{' '}
```
Comment (tetogomez) [inline src/components/VerificationDialog/VerificationDialog.js]: same here

File: src/components/VerificationDialog/VerificationDialog.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+              {verificationDetails.block_num}
+            </Typography>
+            <Typography>
+              <strong>{t('verificationDialog.transactionId')} </strong>{' '}
```
Comment (tetogomez) [inline src/components/VerificationDialog/VerificationDialog.js]: and here

**Review by tetogomez — COMMENTED** [review]
Please check the comments!

**Pull request merged**